public/app/features/explore/QueryRow.tsx

```diff
@@ -76,8 +77,15 @@ export class QueryRow extends PureComponent<QueryRowProps, QueryRowState> {
     this.props.addQueryRow(exploreId, index);
   };
 
-  onClickClearButton = () => {
-    this.onChange(null, true);
+  onClickToggleHiddenQuery = () => {
+    const { exploreId, index, query, queryResponse } = this.props;
+    if (queryResponse.state !== LoadingState.NotStarted) {
```

Comment:
What about moving this inside the queryActions so that the show/hide button is disabled in this case?